packages/dev/core/src/Engines/Extensions/engine.alpha.ts

```diff
@@ -26,92 +27,108 @@ ThinEngine.prototype.setAlphaMode = function (mode: number, noDepthWriteChange:
         return;
     }
 
+    const alphaBlendDisabled = mode === Constants.ALPHA_DISABLE;
+
+    this._alphaState.setAlphaBlend(!alphaBlendDisabled, targetIndex);
+
     switch (mode) {
         case Constants.ALPHA_DISABLE:
-            this._alphaState.alphaBlend = false;
             break;
         case Constants.ALPHA_PREMULTIPLIED:
-            this._alphaState.setAlphaBlendFunctionParameters(this._gl.ONE, this._gl.ONE_MINUS_SRC_ALPHA, this._gl.ONE, this._gl.ONE);
-            this._alphaState.alphaBlend = true;
+            this._alphaState.setAlphaBlendFunctionParameters(this._gl.ONE, this._gl.ONE_MINUS_SRC_ALPHA, this._gl.ONE, this._gl.ONE, targetIndex);
+            this._alphaState.setAlphaEquationParameters(Constants.GL_ALPHA_EQUATION_ADD, Constants.GL_ALPHA_EQUATION_ADD, targetIndex);

```

Comment:
this sounds like a lot of this code could be factorized ?

like init foo = Constants.GL_ALPHA_EQUATION_ADD, later replace with substract min or max in the 4 cases doing it and after the switch `this._alphaState.setAlphaEquationParameters(foo, foo, targetIndex);`

Reply:
I've also moved the code to `alphaCullingState`, to avoid duplication.